src/trio/_tests/test_util.py

```diff
@@ -323,25 +324,36 @@ async def test_raise_single_exception_from_group() -> None:
         [
             ValueError("foo"),
             ValueError("bar"),
-            KeyboardInterrupt("this exc doesn't get reraised"),
+            KeyboardInterrupt("is not reraised but preserve error msg"),
         ],
     )
-    with pytest.raises(KeyboardInterrupt, match=r"^$") as excinfo:
+    with pytest.raises(
+        KeyboardInterrupt,
+        match=r"is not reraised but preserve error msg",
+    ) as excinfo:
         raise_single_exception_from_group(eg_ki)
+
     assert excinfo.value.__cause__ is eg_ki
     assert excinfo.value.__context__ is None
 
-    # and same for SystemExit
+    # and same for SystemExit but verify code too
+    sysexit_exc = SystemExit("preserve error code")
+    sysexit_exc.code = 1  # can't set it in constructor
+
     systemexit_ki = BaseExceptionGroup(
         "",
         [
             ValueError("foo"),
             ValueError("bar"),
-            SystemExit("this exc doesn't get reraised"),
+            sysexit_exc,
         ],
     )
-    with pytest.raises(SystemExit, match=r"^$") as excinfo:
+
+    with pytest.raises(
+        SystemExit, check=lambda e: e.code == 1, match=r"preserve error code"
```

Comment:
Instead of using check and needing a newer pytest version, you could add the check below next to the `excinfo` asserts.

Reply:
Oh I assumed it would just be `excinfo.value.code`, but I don't know the pytest APIs very well

Reply:
it exposes error code as general args[0] field but pytest.raises excinfo doesn't provide access to custom exception fields (check argument example in pytest docs literally checks OSError.errno https://docs.pytest.org/en/latest/_modules/_pytest/raises.html )